File: builder/MessengerNotificationBuilder.java
package builder;

import java.util.List;

public class MessengerNotificationBuilder implements NotificationBuilder {

    private String content;
    private String sender;
    private String recipient;
    private String timestamp;
    private List<String> attachments;
    private String theme;

    public MessengerNotificationBuilder setAttachments(List<String> attachments) {
        this.attachments = attachments;
        return this;
    }

    public MessengerNotificationBuilder setTheme(String theme) {
        this.theme = theme;
        return this;
    }

    @Override
    public MessengerNotificationBuilder setContent(String content) {
        this.content = content;
        return this;
    }

    @Override
    public MessengerNotificationBuilder setSender(String sender) {
        this.sender = sender;
        return this;
    }

    @Override
    public MessengerNotificationBuilder setRecipient(String recipient) {
        this.recipient = recipient;
        return this;
    }

    @Override
    public MessengerNotificationBuilder setTimestamp(String timestamp) {
        this.timestamp = timestamp;
        return this;
    }

    public String getContent() {
        return content;
    }

    public String getSender() {
        return sender;
    }

    public String getRecipient() {
        return recipient;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public List<String> getAttachments() {
        return attachments;
    }

    public String getTheme() {
        return theme;
    }

    public MessengerNotification build() {
        return new MessengerNotification(this);
    }
}

File: C#/Builder/MessengerNotificationBuilder.cs
namespace Builder;

public sealed class MessengerNotificationBuilder : INotificationBuilder
{
    public string? Content { get; private set; }
    public string? Sender { get; private set; }
    public string? Recipient { get; private set; }
    public string? Timestamp { get; private set; }
    public List<string>? Attachments { get; private set; }
    public string? Theme { get; private set; }

    public MessengerNotification Build()
        => new(this);

    public INotificationBuilder SetContent(string content)
    {
        Content = content;
        return this;
    }

    public INotificationBuilder SetSender(string sender)
    {
        Sender = sender;
        return this;
    }

    public INotificationBuilder SetRecipient(string recipient)
    {
        Recipient = recipient;
        return this;
    }

    public INotificationBuilder SetTimestamp(string timestamp)
    {
        Timestamp = timestamp;
        return this;
    }

    public INotificationBuilder SetAttachments(List<string> attachments)
    {
        Attachments = attachments;
        return this;
    }

    public INotificationBuilder SetTheme(string theme)
    {
        Theme = theme;
        return this;
    }
}
